tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -373,14 +361,23 @@ int main(int argc, char **argv)
         struct client_hello_context client_hello_ctx = {.invocations = 0 };
         EXPECT_SUCCESS(s2n_config_set_client_hello_cb(server_config, client_hello_detect_duplicate_calls, &client_hello_ctx));
 
+        /* Force HRR path by sending an empty list of keyshares */
+        EXPECT_SUCCESS(s2n_connection_set_keyshare_by_name_for_testing(client_conn, "none"));
+
         /* Send the first CH message */
         EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
-        EXPECT_SUCCESS(s2n_stuffer_write(&server_conn->handshake.io, &client_conn->handshake.io.blob));
+        EXPECT_SUCCESS(s2n_stuffer_copy(&client_conn->handshake.io, &server_conn->handshake.io,
+                                        s2n_stuffer_data_available(&client_conn->handshake.io)));
 
         /* Receive the CH and send an HRR, which will execute the HRR code paths */
         EXPECT_EQUAL(client_hello_ctx.invocations, 0);
         EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
         EXPECT_EQUAL(client_hello_ctx.invocations, 1);
+
+        EXPECT_TRUE(s2n_is_hello_retry_handshake(server_conn));
+        EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(server_conn, 1));
```

Comment:
Why do you need to call `s2n_set_connection_hello_retry_flags`? Doesn't that negate the value of this test?

Reply:
`s2n_set_connection_hello_retry_flags` is called after checking whether the handshake is set correctly so it is not negating the test. The reason why `s2n_set_connection_hello_retry_flags` needs to be called is to advance the message number to `HELLO_RETRY_MSG`.